### PR TITLE
Distance sampling

### PR DESCRIPTION
**Pull request by wlangera**

### CODE REVIEW

File: src/R/plot_detection_curve.R [hunk 1]
```diff
@@ -266,24 +278,34 @@ plot_detection_curve <- function(dist_model, design_mat = NULL, labels = NULL,
...
+              legend.background = element_rect(fill = "white", color = "white"))
     }
   }
 
+  if (show_data) {
+    distance_data <- data.frame(dist = sort(unique(dist_model$ddf$data$distance)))
+    out <- out +
+      geom_point(data = distance_data, aes(x = dist,
+                                           y = rep(0, nrow(distance_data))),
+                 shape = "|", col = "brown", size = 3)
```
Comment (hansvancalster) [inline src/R/plot_detection_curve.R:306]: This works, but ggplot has a geom for this. See https://ggplot2.tidyverse.org/reference/geom_rug.html

File: src/R/summarize_ds_models2.R [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    }
+  }
+
+
+  ## checking
+  # can't compare models with different truncations
+  r_truncs <- unlist(lapply(models, function(x) x$ddf$meta.data$width))
+  l_truncs <- unlist(lapply(models, function(x) x$ddf$meta.data$left))
+  if(!all(abs(c(r_truncs-mean(r_truncs),
+                l_truncs-mean(l_truncs))) < 1e-8)){
```
Comment (hansvancalster) [inline src/R/summarize_ds_models2.R:35]:
```suggestion
  if(!all(abs(c(r_truncs - mean(r_truncs),
                l_truncs - mean(l_truncs))) < 1e-8)){
```

File: src/R/summarize_ds_models2.R [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      average.p <- summ$ds$average.p
+      average.p.se <- summ$ds$average.p.se
+    }
+
+
+    # handle (uniform) no formula case
+    formula <- model$ddf$ds$aux$ddfobj$scale$formula
+    if(is.null(formula)) formula <- NA
+
+    desc <- gsub(" key function","",Distance:::model.description(model$ddf))
```
Comment (hansvancalster) [inline src/R/summarize_ds_models2.R:76]:
```suggestion
    desc <- gsub(" key function", "", Distance:::model.description(model$ddf))
```

File: src/R/summarize_ds_models2.R [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+  # applying that to all the models then putting it into a data.frame
+  res <- as.data.frame(t(as.data.frame(lapply(models, extract_model_data))),
+                       stringsAsFactors=FALSE)
+
+  if(output == "latex"){
+    model_names <- gsub("_", '\\_', model_names)
+    #model_names <- paste0("\\texttt{", model_names, "}")
+    res <- cbind.data.frame(model_names, res)
+  }else if(output=="plain"){
```
Comment (hansvancalster) [inline src/R/summarize_ds_models2.R:102]:
```suggestion
  } else if (output == "plain") {
```

File: src/R/summarize_ds_models2.R [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+             model$ddf$criterion
+    )
+    return(ret)
+  }
+
+  # applying that to all the models then putting it into a data.frame
+  res <- as.data.frame(t(as.data.frame(lapply(models, extract_model_data))),
+                       stringsAsFactors=FALSE)
+
+  if(output == "latex"){
```
Comment (hansvancalster) [inline src/R/summarize_ds_models2.R:98]:
```suggestion
  if (output == "latex") {
```

File: src/R/summarize_ds_models2.R [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    model_names <- gsub("_", '\\_', model_names)
+    #model_names <- paste0("\\texttt{", model_names, "}")
+    res <- cbind.data.frame(model_names, res)
+  }else if(output=="plain"){
+    res <- cbind.data.frame(model_names, res)
+  }else{
```
Comment (hansvancalster) [inline src/R/summarize_ds_models2.R:104]:
```suggestion
  } else {
```

File: src/R/summarize_ds_models2.R [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    colnames(res) <- c("Model",
+                       "Key function",
+                       "Formula",
+                       gof_name,
+                       "Average detectability",
+                       "se(Average detectability)",
+                       "AIC")
+  }else{
+    stop("Invalid output format")
+  }
```
Comment (hansvancalster) [inline src/R/summarize_ds_models2.R:139]:
```suggestion
  if (all(binned)) {
    gof_name <- "Chi^2 p-value"
    gof_latexname <- "$\\chi^2$ $p$-value"
  } else {
    gof_name <- "C-vM $p$-value"
    gof_latexname <- "C-vM p-value"
  }

  # giving the columns names
  if (output == "latex") {
    colnames(res) <- c("Model",
                       "Key function",
                       "Formula",
                       gof_latexname,
                       "$\\hat{P_a}$",
                       "se($\\hat{P_a}$)",
                       "AIC")
  } else if (output=="plain") {
    colnames(res) <- c("Model",
                       "Key function",
                       "Formula",
                       gof_name,
                       "Average detectability",
                       "se(Average detectability)",
                       "AIC")
  } else {
    stop("Invalid output format")
  }
```

File: src/R/summarize_ds_models2.R [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+  # remove row names
+  rownames(res) <- NULL
+
+  # creating a new column for the AIC difference to the best model
+  if(output == "latex"){
+    res[["$\\Delta$AIC"]] <- res$AIC - min(res$AIC, na.rm=TRUE)
+  }else if(output=="plain"){
+    res[["Delta AIC"]] <- res$AIC - min(res$AIC, na.rm=TRUE)
+  }
```
Comment (hansvancalster) [inline src/R/summarize_ds_models2.R:148]:
```suggestion
  if (output == "latex") {
    res[["$\\Delta$AIC"]] <- res$AIC - min(res$AIC, na.rm=TRUE)
  } else if (output == "plain") {
    res[["Delta AIC"]] <- res$AIC - min(res$AIC, na.rm=TRUE)
  }
```

File: src/markdown/mas/distance_sampling.Rmd [hunk 9]
```diff
@@ -0,0 +1,4153 @@
...
+```{r}
+roofvogels <- c("Blauwe Kiekendief", "Boomvalk", "Bruine Kiekendief", "Buizerd", 
+                "Grauwe Kiekendief", "Ransuil", "Slechtvalk", "Sperwer", 
+                "Steenuil", "Steppekiekendief", "Torenvalk", "Velduil", 
+                "Wespendief", "Zwarte Wouw")
+```
+
+We maken een dataframe `mas_observations` waarbij we alle waarnemingen van vogels selecteren en filteren op broedcode \> 0, geen afstanden \> 300 m en dat ze binnen telperiodes vallen.
+We behouden ook enkel de plots die tot `design_2018_2022_mas` behoren.
+We voegen ook Haas toe en roofvogels (`r roofvogels`) zonder te filteren op broedcode.
```
Comment (hansvancalster) [inline src/markdown/mas/distance_sampling.Rmd:1834]: Na te vragen bij Johannes of de kraaiachtigen ook zoals de roofvogels behandeld moeten worden.

File: src/markdown/mas/distance_sampling.Rmd [hunk 10]
```diff
@@ -0,0 +1,4153 @@
...
+
+do.call(rbind, list_2022) %>%
+  kable()
+```
+
+## Model specificatie distance sampling
+
+### 2018-2022
+
+Voor de formulatie van de detectiefunctie beschouwen we half-normal en hazard-rate sleutelfuncties die we laten afhangen van covariaten regio en jaar. We fitten modellen zonder covariaten, met alle combinaties van covariaten en beschouwen ook tweewegsinteracties tussen de covariaten. In de modellen zonder covariaten kunnen aanpassingstermen toegevoegd worden om een betere fit te krijgen tussen de curve en de data. Om te hoge flexibiliteit van dergelijke detectiefunctiemodellen te beperken beperken we het aantal aanpassingstermen tot maximaal 2. De modellen worden lokaal opgeslaan zodat ze maar eenmaal moeten gefit worden.
```
Comment (hansvancalster) [inline src/markdown/mas/distance_sampling.Rmd:2400]:
```suggestion
Voor de formulatie van de detectiefunctie beschouwen we half-normal en hazard-rate sleutelfuncties die we laten afhangen van covariaten regio en jaar. We fitten modellen zonder covariaten, met alle combinaties van covariaten en beschouwen ook tweewegsinteracties tussen de covariaten. In de modellen zonder covariaten kunnen aanpassingstermen toegevoegd worden om een betere fit te krijgen tussen de curve en de data. Om te hoge flexibiliteit van dergelijke detectiefunctiemodellen te beperken, beperken we het aantal aanpassingstermen tot maximaal 2. De modellen worden lokaal opgeslagen zodat ze maar eenmaal moeten gefit worden.
```

File: src/markdown/mas/distance_sampling.Rmd [hunk 11]
```diff
@@ -0,0 +1,4153 @@
...
+  # Put models and model names in list 
+  all_models <- c(all_models, list(out_models))
+  all_model_names <- c(all_model_names, spec)
+}
+names(all_models) <- all_model_names
+```
+
+### 2022
+
+Voor de formulatie van de detectiefunctie beschouwen we half-normal en hazard-rate sleutelfuncties die we laten afhangen van covariaten regio, openheid en sbp. We fitten modellen zonder covariaten, met alle combinaties van covariaten en beschouwen enkel tweewegsinteracties tussen de covariaten. In de modellen zonder covariaten kunnen aanpassingstermen toegevoegd worden om een betere fit te krijgen tussen de curve en de data. Om te hoge flexibiliteit van dergelijke detectiefunctiemodellen te beperken beperken we het aantal aanpassingstermen tot maximaal 2. De modellen worden lokaal opgeslaan zodat ze maar eenmaal moeten gefit worden.
```
Comment (hansvancalster) [inline src/markdown/mas/distance_sampling.Rmd:2527]:
```suggestion
Voor de formulatie van de detectiefunctie beschouwen we half-normal en hazard-rate sleutelfuncties die we laten afhangen van covariaten regio, openheid en sbp. We fitten modellen zonder covariaten, met alle combinaties van covariaten en beschouwen enkel tweewegsinteracties tussen de covariaten. In de modellen zonder covariaten kunnen aanpassingstermen toegevoegd worden om een betere fit te krijgen tussen de curve en de data. Om te hoge flexibiliteit van dergelijke detectiefunctiemodellen te beperken, beperken we het aantal aanpassingstermen tot maximaal 2. De modellen worden lokaal opgeslagen zodat ze maar eenmaal moeten gefit worden.
```

**Review by hansvancalster — APPROVED** [review]
Indrukwekkend werkstuk 🚀

File: src/markdown/mas/distance_sampling.Rmd (hunk 9 above)
Comment (wlangera) [reply src/markdown/mas/distance_sampling.Rmd:1834]: Ok, zal ik doen

File: src/R/plot_detection_curve.R (hunk 1 above)
Comment (wlangera) [reply src/R/plot_detection_curve.R:306]: For some reason I could not find that geom! I just tested it and I think the position looks better with geom_point (lines too low with geom_rug).

**Pull request merged**